File: app/src/main/java/com/example/learnenglish/ui/main/MainActivity.java
package com.example.learnenglish.ui.main;

import android.os.Bundle;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;

import com.example.learnenglish.R;
import com.example.learnenglish.ui.communication.test.VocabularyFragment;
import com.example.learnenglish.ui.vocabulary.AlphabetFragment;
import com.example.learnenglish.ui.vocabulary.SearchFragment;
import com.example.learnenglish.ui.vocabulary.TestFragment;
import com.google.android.material.navigation.NavigationView;

public class MainActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener {
    private DrawerLayout drawer;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar =  findViewById(R.id.toolbar);
        //setSupportActionBar(toolbar);
        drawer = findViewById(R.id.draw_layout);

        NavigationView navigationView = findViewById(R.id.nav_view);
        navigationView.setNavigationItemSelectedListener(this);

        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(this,drawer,toolbar,R.string.navigation_drawer_open,R.string.navigation_drawer_close);
        drawer.addDrawerListener(toggle);
        toggle.syncState();
        if(savedInstanceState==null){
            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,new VocabularyFragment()).commit();
            navigationView.setCheckedItem(R.id.nav_vocabulary);
        }
        String a= "qwe";
        if (a.equalsIgnoreCase("QWE")) {

        }
    }

    @Override
    public void onBackPressed() {
        if (drawer.isDrawerOpen(GravityCompat.START)){
            drawer.closeDrawer(GravityCompat.START);
        }else super.onBackPressed();
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()){
            case R.id.nav_vocabulary:
                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,new VocabularyFragment()).commit();
                break;
            case R.id.nav_learnalphabet:
                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,new AlphabetFragment()).commit();
                break;
            case R.id.nav_search:
                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,new SearchFragment()).commit();
                break;
            case R.id.nav_test:
                TestFragment testFragment= new TestFragment();
                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,testFragment,testFragment.getTag()).commit();
                break;
            case R.id.nav_share:
                Toast.makeText(this,"Share",Toast.LENGTH_SHORT).show();
                break;
            case R.id.nav_setting:
                Toast.makeText(this,"Setting",Toast.LENGTH_SHORT).show();
                break;
        }
        drawer.closeDrawer(GravityCompat.START);
        return true;
    }


}

File: app/src/main/java/com/example/learnenglish/ui/main/MainActivity.kt
package com.example.learnenglish.ui.main

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.learnenglish.R
import com.example.learnenglish.database.VocabularyDatabase
import com.example.learnenglish.ui.authentication.profile.ProfileActivity
import com.example.learnenglish.ui.authentication.signin.SignInActivity
import com.example.learnenglish.ui.communication.CommunicationTopicActivity
import com.example.learnenglish.ui.grammar.GrammarTopicActivity
import com.example.learnenglish.ui.practice.PracticeTopicActivity
import com.example.learnenglish.ui.translate.TranslateActivity
import com.example.learnenglish.ui.vocabulary.topic.VocabularyActivity
import com.google.firebase.auth.ktx.auth
import com.google.firebase.ktx.Firebase
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.include_toolbar.*
import java.io.File

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private lateinit var vocabularyDatabase: VocabularyDatabase

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setSupportActionBar(toolbar)
        val actionBar = supportActionBar
        actionBar?.let {
            tvTitleToolbar.text = title
            it.title = ""
        }

        vocabularyDatabase = VocabularyDatabase(this)
        val database: File = getDatabasePath(VocabularyDatabase.DBNAME)
        if (!database.exists()) {
            vocabularyDatabase.readableDatabase
            if (vocabularyDatabase.copyDatabase(this)) {
                //Toast.makeText(this, "Copy database success", Toast.LENGTH_SHORT).show()
            } else {
                //Toast.makeText(this, "Copy data error", Toast.LENGTH_SHORT).show()
            }
        }

        cvVoca.setOnClickListener(this)
        cvGrammar.setOnClickListener(this)
        cvCommunication.setOnClickListener(this)
        cvTranslate.setOnClickListener(this)
        cvPractice.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            cvVoca.id -> {
                startActivity(Intent(this, VocabularyActivity::class.java))
            }

            cvGrammar.id -> {
                startActivity(Intent(this, GrammarTopicActivity::class.java))
            }

            cvCommunication.id -> {
                startActivity(Intent(this, CommunicationTopicActivity::class.java))
            }

            cvTranslate.id -> {
                startActivity(Intent(this, TranslateActivity::class.java))
            }

            cvPractice.id -> {
                startActivity(Intent(this, PracticeTopicActivity::class.java))
            }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when(item.itemId) {
            R.id.action_login -> {
                if (Firebase.auth.currentUser?.isAnonymous == null) {
                    startActivity(Intent(this, SignInActivity::class.java))
                } else {
                    startActivity(Intent(this, ProfileActivity::class.java))
                }
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}